pkgs/tools/networking/slack-cli/default.nix

```diff
@@ -0,0 +1,27 @@
+# slack-cli must be configured using the SLACK_CLI_TOKEN environment
+# variable. Using `slack init` will not work because it tries to write
+# to the Nix store.
+
+{ stdenv, lib, fetchFromGitHub, makeWrapper, curl, jq }:
+
+stdenv.mkDerivation rec {
+  name = "slack-cli";
+  version = "0.18.0";
+
+  src = fetchFromGitHub {
+    owner = "rockymadden";
+    repo = "slack-cli";
+    rev = "v${version}";
+    sha256 = "022yr3cpfg0v7cxi62zzk08vp0l3w851qpfh6amyfgjiynnfyddl";
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  dontBuild = true;
```

Comment:
Where do you actually use `make`?

Reply:
The `makeWrapper` dependency is for the `installPhase` where `wrapProgram` is called.

Reply:
Ooh that is a confusing name. I assumed it was a wrapper for make :grinning:

It should be a `nativeBuildInput` then. The difference is that `nativeBuildInputs` are used exclusively during packaging. The distinction is important for cross-compilation.

Alternatively you could leave it out of any `buildInput`s and just use `${makeWrapper}/bin/wrapProgram` in `installPhase`.

Reply:
> fwiw

Yes that is because the distinction is relatively recent, many contributors aren't aware of it and it is not enforced in regular compilation (yet). But the right way to do it is to always put `makeWrapper` in `nativeBuildInputs`. Maybe some point in the future we will enforce the distinction for regular compilation.

For example when you want to compile for a raspberry pi on your regular x86 desktop, you will need a x86 version of `makeWrapper` but aarch versions of all the runtime dependencies.

> This wouldn't work.

Huh, I wonder how it works when it is in `buildInputs`. Anyways, that was a misunderstanding on my side. The way you do it currently is fine (except native vs. regular build inputs).

Reply:
Cool, thanks for the explanation :)